lib/quadquizaminos_web/live/tetris_live.ex

```diff
@@ -619,19 +621,35 @@ defmodule QuadquizaminosWeb.TetrisLive do
   end
 
   defp move_block(socket, x, y, block_coordinates, true = _adding_block, true = _moving_block) do
-    {x, y} = parse_to_integer(x, y)
-    {x1, y1, color} = block_coordinates
-    powers = socket.assigns.powers -- [:moveblock]
-
-    bottom =
-      socket.assigns.bottom |> Map.delete({x1, y1}) |> Map.merge(%{{x, y} => {x, y, color}})
+    # check if the coordinates are part of bottom, if they're not return the socket, if they're apply transformation
+    # and return socket
+    ycoordinate =
+      socket.assigns.bottom
+      |> Map.keys()
+      |> Enum.map(fn value -> elem(value, 1) end)
+      |> Enum.map(fn c -> Integer.to_string(c) end)
+
+    if y in ycoordinate do
+      {x, y} = parse_to_integer(x, y)
+      {x1, y1, color} = block_coordinates
+      powers = socket.assigns.powers -- [:moveblock]
 
-    assign(socket,
-      bottom: bottom,
-      powers: powers,
-      adding_block: false,
-      moving_block: false
-    )
+      bottom =
+        socket.assigns.bottom
+        |> Map.delete({x1, y1})
+        |> Map.merge(%{{x, y} => {x, y, color}})
+        |> IO.inspect()
+
+      assign(socket,
+        bottom: bottom,
+        powers: powers,
+        adding_block: false,
+        moving_block: false
+      )
+    else
+      powers = socket.assigns.powers -- [:moveblock]
```

Comment:
Since we never successfully used the power up we shold not remove it from the socket

Reply:
I agree with this